src/function/FunctionRestriction.ts

```diff
@@ -1,63 +1,47 @@
+import {ProjectElements, ProjectLinks, ProjectSettings, VocabularyElements} from "../config/Variables";
+import {Restriction} from "../datatypes/Restriction";
+import {getNewLink} from "./FunctionGetVars";
+import {LinkType} from "../config/Enum";
+import {addLink} from "./FunctionCreateVars";
 
-export function createRestriction(obj: { [key: string]: any }, iri: string, restriction: string, onProperty: string, target: { type: string, value: string }) {
-	if (target.type !== "bnode" && (restriction in Restrictions)) {
-		let newRestriction = new RestrictionObject(restriction, onProperty, target.value);
+export function createRestriction(obj: { [key: string]: any }, iri: string, restriction: string, onProperty: string, target: { type: string, value: string }, onClass?: string) {
+	if (target.type !== "bnode") {
+		let newRestriction = new Restriction(restriction, onProperty, target.value, onClass);
 		for (let rest of obj[iri].restrictions) {
 			if (rest.target === newRestriction.target
 				&& rest.restriction === newRestriction.restriction
-				&& rest.onProperty === newRestriction.onProperty) {
+				&& rest.onProperty === newRestriction.onProperty
+				&& rest.onClass === newRestriction.onClass) {
 				return;
 			}
 		}
 		obj[iri].restrictions.push(newRestriction);
-	}
-}
-
-export function initRestrictions() {
-	for (let iri in VocabularyElements) {
-		for (let restriction of VocabularyElements[iri].restrictions) {
-			restriction.initRestriction(iri);
-		}
-		let domain = VocabularyElements[iri].domain;
-		let range = VocabularyElements[iri].range;
-		if (typeof domain === "string" && typeof range === "string" && VocabularyElements[domain]) {
-			VocabularyElements[domain].connections.push(new ConnectionObject(iri, range, false));
-		}
+		obj[iri].restrictions.sort((a: Restriction, b: Restriction) => a.restriction.localeCompare(b.restriction));
 	}
 }
 
 export function initConnections(): string[] {
-	let result: string[] = [];
+	let linksToPush = [];
 	for (let iri in VocabularyElements) {
-		for (let connection of VocabularyElements[iri].connections) {
-			let conn = connection.initConnection(iri);
-			if (conn) result.push(conn);
+		for (let restriction of VocabularyElements[iri].restrictions) {
+			let newLink = restriction.initRestriction(iri);
+			if (newLink) linksToPush.push(newLink);
 		}
-	}
-	return result;
-}
 
-export function getRestrictionsAsJSON(iri: string) {
-	let result: {}[] = [];
-	for (let restriction of VocabularyElements[iri].restrictions) {
-		for (let id in ProjectLinks) {
-			if (ProjectLinks[id].iri === iri && ProjectElements[ProjectLinks[id].source].iri) {
-				restriction.saveRestriction(ProjectElements[ProjectLinks[id].source].iri);
-				break;
+		for (let subClassOf of VocabularyElements[iri].subClassOf) {
+			if (subClassOf in VocabularyElements) {
+				let domainID = Object.keys(ProjectElements).find(element => ProjectElements[element].iri === iri);
```

Comment:
There is this pattern that is copied on multiple places:
`Object.keys(ProjectElements).find(element => ProjectElements[element].iri === XXXXX);`

It could be worth refactor it to a separate utility method, e.g.
`const findElementByIri = (iri: string) => Object.keys(ProjectElements).find(element => ProjectElements[element].iri === iri)`
and then just do something like
`const domainID = findElementByIri(iri)`
It would improve readability a lot!

Reply:
Implemented. Thank you for the comment.